include/RED4ext/Api/v1/GameState.hpp

```diff
@@ -0,0 +1,82 @@
+#pragma once
+
+namespace RED4ext
+{
+struct CGameApplication;
+
+namespace v1
+{
+/**
+ * @brief Result type for game state hooks.
+ */
+enum class EGameStateResult : uint8_t
+{
+    Running,  /**< Hook has not yet finished. Until the state changes, hook continues to operate. */
+    Finished, /**< Hook has finished. When signaled, hook is detached. */
+    Observing /**< Hook is observing, not running a task. Hook won't get detached until shutdown. */
```

Comment:
Similar to "Running", the most important property of "Observing" is that it doesn't prevent game state transition (while still executing).